libs/k8s-client/tests/ApiClient/BaseClusterApiClientTestCase.php

```diff
@@ -0,0 +1,325 @@
+        self::assertSame($resourceToCreate->metadata->name, $createdResource->metadata->name);
+
+        $result = $this->baseApiClient->list([
+            'labelSelector' => sprintf('%s=%s', self::getTestResourcesLabelName(), (string) getenv('K8S_NAMESPACE')),
+        ]);
+        assert(is_object($result) && property_exists($result, 'items'));
+
+        $this->assertResultItems([$createdResource->metadata->name], $result->items);
+    }
+
+    public function testCreateResourceWithDuplicateNameThrowsException(): void
+    {
+        $resourceToCreate = $this->createResource([
+            'name' => self::getTestResourceNameWithNamespacePrefix('test-resource-1'),
+            'labels' => [
+                'app' => 'test-1',
+                'k8s-client-tests-namespace' => (string) getenv('K8S_NAMESPACE'),
+            ],
+        ]);
+
+        $this->apiClient->create($resourceToCreate);
+
+        $this->expectException(ResourceAlreadyExistsException::class);
+        $this->expectExceptionMessage('Resource already exists:');
+
+        $this->apiClient->create($resourceToCreate);
+    }
+
+    public function testDeleteResource(): void
+    {
+        $this->baseApiClient->create($this->createResource([
+            'name' => self::getTestResourceNameWithNamespacePrefix('test-resource-1'),
+            'labels' => [
+                'app' => 'test-1',
+                'k8s-client-tests-namespace' => (string) getenv('K8S_NAMESPACE'),
+            ],
+        ]));
+        $this->baseApiClient->create($this->createResource([
+            'name' => self::getTestResourceNameWithNamespacePrefix('test-resource-2'),
+            'labels' => [
+                'app' => 'test-2',
+                'k8s-client-tests-namespace' => (string) getenv('K8S_NAMESPACE'),
+            ],
+        ]));
+
+        // delete the resource
+        $this->apiClient->delete(self::getTestResourceNameWithNamespacePrefix('test-resource-1'));
+        $this->waitWhileResourceExists(self::getTestResourceNameWithNamespacePrefix('test-resource-1'));
+
+        // check the other resource was not deleted
+        $this->apiClient->get(self::getTestResourceNameWithNamespacePrefix('test-resource-2'));
+
+        $this->expectNotToPerformAssertions();
+    }
+
+    public function testDeleteNotExistingResourceThrowsException(): void
+    {
+        $this->expectException(ResourceNotFoundException::class);
+        $this->expectExceptionMessage('Resource not found:');
+
+        $this->apiClient->delete(self::getTestResourceNameWithNamespacePrefix('test-resource-1'));
+    }
+
+    public function testDeleteCollection(): void
+    {
+        $this->baseApiClient->create($this->createResource([
+            'name' => self::getTestResourceNameWithNamespacePrefix('test-resource-11'),
+            'labels' => [
+                'app' => 'test-1',
+                'k8s-client-tests-namespace' => (string) getenv('K8S_NAMESPACE'),
```

Comment:
V `BaseNamespaceApiClientTestCase` se misto `'k8s-client-tests-namespace'` pouziva `self::getTestResourcesLabelName()`. Nechces to pouzit i v tehle class? Sam si to tam zavadel :wink:

Reply:
fixed